File: app/src/main/java/com/example/beerhive/beerlist/BeerPresenter.java
package com.example.beerhive.beerlist;

import com.example.beerhive.network.model.BeerResponse;

import java.util.List;

public class BeerPresenter implements BeerContract.UserActionsListener {

    private BeerRepository beerRepository;
    private BeerContract.BeerView view;

    public BeerPresenter(BeerContract.BeerView view) {
        this.view = view;
        beerRepository = new BeerRepositoryImplentation();
    }

    @Override
    public void loadBeerList() {
        view.showProgressDialog();

        beerRepository.loadBeerListFromNetwork(new BeerRepository.BeerLoaderCallback() {
            @Override
            public void onBeerListLoaded(List<BeerResponse> beerResponseList) {
                view.displayBeerList(beerResponseList);
                view.dismissProgressDialog();
            }

            @Override
            public void onErrorOccurred() {
                view.dismissProgressDialog();
                view.showErrorMesage();
            }
        });
    }
}

File: app/src/main/java/com/example/beerhive/beerlist/BeerPresenter.kt
package com.example.beerhive.beerlist

import com.example.beerhive.beerlist.BeerContract.BeerView
import com.example.beerhive.beerlist.BeerContract.UserActionsListener
import com.example.beerhive.beerlist.BeerRepository.BeerLoaderCallback
import com.example.beerhive.network.model.BeerResponse

class BeerPresenter(private val view: BeerView) : UserActionsListener {
    private val beerRepository: BeerRepository
    override fun loadBeerList() {
        view.showProgressDialog()
        beerRepository.loadBeerListFromNetwork(object : BeerLoaderCallback {
            override fun onBeerListLoaded(beerResponseList: MutableList<BeerResponse?>?) {
                view.displayBeerList(beerResponseList)
                view.dismissProgressDialog()
            }

            override fun onErrorOccurred() {
                view.dismissProgressDialog()
                view.showErrorMessage()
            }
        })
    }

    init {
        beerRepository = BeerRepositoryImplementation()
    }
}